### ISSUE TITLE
amp-analytics integration test idea

### ISSUE THREAD

**Issue by lannka** [ampproject/amphtml#19211]
We built `describes.integration` as an integration test lib to check of a XHR request from an AMP extension. However, it does not support complicated use case. Imagine the following rendering flow:

1. AMP runtime instantiated amp-ad element
2. amp-ad made ad request, which returned an AMP ad
3. the returned AMP ad contains amp-analytics element
4. the amp-analytics element is configured to inject an xorigin iframe
5. amp-analytics start sending data to it

All existing testing infra AMP has cannot fully handle a test that verifies the xorigin iframe created in step 4 is receiving correct data. 

Here I propose a test-server log based solution. The idea roughly goes like:
1. we compose a valid AMP page that does exactly the above use case
2. we have karma load that AMP page, wait for full page load + a few more seconds, then close 
3. test-server logs the HTTP requests in a local file
4. run a script checking the log file contains expected rows (comparing to a golden file)

Many details need to be ironed out:
- potential flakiness that can be caused by the waiting on step 2.
- Logged HTTP requests might not always be in order. Step 4 needs provide various asserts function to meet test needs.
- anyway to introduce user interaction to the page, for example scroll & click?

 thoughts?

/cc @rsimha @zhouyx @torch2424 @calebcordry @dvoytenko @cramforce @jpettitt

**Comment by lannka** [comment]
@estherkim I know you're looking at Selenium based integration tests. Do you think it would cover the use case as described above? We are especially interested to verify the HTTP requests sent from AMP runtime and received from the server.

**Comment by estherkim** [comment]
@lannka I'm not sure yet - give me a few days to understand how `describes.integration` works and how I might be able to capture the HTTP requests using our existing infra for integration tests. Might need your assistance in building out the test AMP page.

**Comment by zhouyx** [comment]
Another thing to test with is when does a request gets sent. Would that be possible?

**Comment by lannka** [comment]
After a series of improvements (#19605, #19613, #20037, #20062) to `describes.integration` during the fixit week, I think most of the requirements here are already covered. Closing this now.
